### PR TITLE
Add some stuff to the apps elasticsearch mapping

### PR DESCRIPTION
**Pull request by esoergel**
First commit formats the file but has no actual changes.  I'll roll this out manually on prod, india, and ICDS.  I'm guessing there are few enough docs on the other envs that this can just go out on deploy (that's probably true for ICDS too, but I might as well).

Any other wishlist app mapping changes we should roll in to this?  The mapping is huge, which is kinda silly considering how infrequently we actually use it.

@orangejenny @proteusvacuum

### CODE REVIEW

File: corehq/pillows/mappings/app_mapping.py [hunk 1]
```diff
@@ -117,7 +117,7 @@
         "use_custom_suite": {"type": "boolean"},
         "cached_properties": {"type": "object", "dynamic": False},
         "modules": {
-            "type": "object",
+            "type": "nested",
```
Comment (esoergel) [inline corehq/pillows/mappings/app_mapping.py]: I do need to rewrite one query to support this, I think.  Working on that now

Comment (esoergel) [reply corehq/pillows/mappings/app_mapping.py]: https://github.com/dimagi/commcare-hq/pull/23079/commits/0f44e83eb97b202bbb55e75bd22d20940753b549

**Review by proteusvacuum — COMMENTED** [review]
1. do we still need the `_all` field? https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-all-field.html . It's removed in newer versions of ES by default.
2. It might be nice to add an `@indexed_on` field like so: https://github.com/dimagi/commcare-hq/blob/c810900f1e098565f1121b7cd926a3921c021637/corehq/pillows/mappings/case_search_mapping.json#L13-L16 (and also add to the pillow)

File: corehq/pillows/mappings/app_mapping.py [hunk 2]
```diff
@@ -4,321 +4,460 @@
 from pillowtop.es_utils import ElasticsearchIndexInfo
 
 APP_INDEX = es_index("hqapps_2017-05-22_1426")
-APP_MAPPING={'_meta': {'created': None},
```
[the remark this answers is not in the document]
Comment (esoergel) [reply corehq/pillows/mappings/app_mapping.py]: Haha nice, I was thinking about writing something like this, didn't realize we already had it.  I'll make that switch.

**Comment by esoergel** [comment]
> do we still need the _all field? https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-all-field.html . It's removed in newer versions of ES by default.

@proteusvacuum where do you mean?  I see an `_all` field in some other mappings, but not this one.

> It might be nice to add an @indexed_on field

Agreed, will do

**Comment by proteusvacuum** [comment]
`_all` is included in mappings by default, but it just duplicates all the data in the doc into a very long string (essentially doubling the size of each document in our index). Docs [here](https://www.elastic.co/guide/en/elasticsearch/reference/1.7/mapping-all-field.html)

You can disable this field like this: https://github.com/dimagi/commcare-hq/blob/c810900f1e098565f1121b7cd926a3921c021637/corehq/pillows/mappings/case_search_mapping.json#L9-L11

**Comment by proteusvacuum** [comment]
Also, you should remove all references to `multi_field`: https://www.elastic.co/guide/en/elasticsearch/reference/1.7/_multi_fields.html

**Comment by emord** [comment]
@proteusvacuum I think the "appstore" (app exchange?) is querying against _all https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/appstore/views.py#L268

**Comment by proteusvacuum** [comment]
@emord nice catch. That seems like total overkill for that page. We should change that to just search descriptions and names. I don't think it's necessary to be searching the whole app doc for stuff there. Maybe it's out of scope for this PR, but we'll need to remove that field at some point on all indexes at before upgrading.

**Comment by esoergel** [comment]
I started looking in to switching this to a `multi_match` query, but it looks like that actually happens in the domains index, _not_ applications.  I guessed this when I saw [`{'doc_type': "Domain"}`](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/appstore/views.py#L257).  Oddly enough, this appears to happen _implicitly_ [here](https://github.com/dimagi/commcare-hq/blob/d3f2f881e486f72f0ba8195371dc17ced0b5d0a4/corehq/elastic.py#L439-L439).

I do all see `_all` used here: https://github.com/dimagi/commcare-hq/blob/d3f2f881e486f72f0ba8195371dc17ced0b5d0a4/corehq/apps/hqadmin/reports.py#L639-L639
Just added a commit to remove that usage and one to remove _all from the index

**Pull request merged**